### PR TITLE
add remark

### PR DESCRIPTION
**Pull request by ngchisern**
* adds 'remark' field to the Person class
* update add command to support remark field 
* update edit command to support remark field 
* update the storage to store the remark field

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Remark.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's name in the address book.
```
Comment (GnohChengYi) [inline src/main/java/seedu/address/model/person/Remark.java:7]:
```suggestion
 * Represents a Person's remark in the address book.
```

File: src/main/java/seedu/address/model/person/Remark.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     * @param remark A valid remark.
+     */
+    public Remark(String remark) {
+        requireNonNull(remark);
+        checkArgument(isValidRemark(remark), MESSAGE_CONSTRAINTS);
+        value = remark;
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
```
Comment (GnohChengYi) [inline src/main/java/seedu/address/model/person/Remark.java:29]:
```suggestion
     * Returns true if a given string is a valid remark.
```

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -45,13 +48,16 @@
...
+    public static final String REMARK_DESC_AMY = " " + PREFIX_REMARK + VALID_REMARK_AMY;
+    public static final String REMARK_DESC_BOB = " " + PREFIX_REMARK + VALID_REMARK_BOB;
     public static final String TAG_DESC_FRIEND = " " + PREFIX_TAG + VALID_TAG_FRIEND;
     public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
 
     public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "James&"; // '&' not allowed in names
     public static final String INVALID_PHONE_DESC = " " + PREFIX_PHONE + "911a"; // 'a' not allowed in phones
     public static final String INVALID_EMAIL_DESC = " " + PREFIX_EMAIL + "bob!yahoo"; // missing '@' symbol
     public static final String INVALID_ADDRESS_DESC = " " + PREFIX_ADDRESS; // empty string not allowed for addresses
+    public static final String INVALID_REMARK_DESC = " " + PREFIX_REMARK + "^happy"; // '^' not allowed in tags
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:60]:
```suggestion
    public static final String INVALID_REMARK_DESC = " " + PREFIX_REMARK + "^happy"; // '^' not allowed in remarks
```

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java [hunk 4]
```diff
@@ -51,6 +52,10 @@ public void equals() {
         editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withAddress(VALID_ADDRESS_BOB).build();
         assertFalse(DESC_AMY.equals(editedAmy));
 
+        //different remark -> false
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:55]:
```suggestion
        // different remark -> false
```

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 5]
```diff
@@ -49,36 +51,40 @@ public void parse_allFieldsPresent_success() {
...
         assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_AMY + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ADDRESS_DESC_BOB + REMARK_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
 
         // multiple addresses - last address accepted
         assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_AMY
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ADDRESS_DESC_BOB + REMARK_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+
+        //multiple remarks - last remark accepted
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:72]:
```suggestion
        // multiple remarks - last remark accepted
```

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 6]
```diff
@@ -85,6 +89,7 @@ public void parse_invalidValue_failure() {
         assertParseFailure(parser, "1" + INVALID_PHONE_DESC, Phone.MESSAGE_CONSTRAINTS); // invalid phone
         assertParseFailure(parser, "1" + INVALID_EMAIL_DESC, Email.MESSAGE_CONSTRAINTS); // invalid email
         assertParseFailure(parser, "1" + INVALID_ADDRESS_DESC, Address.MESSAGE_CONSTRAINTS); // invalid address
+        assertParseFailure(parser, "1" + INVALID_REMARK_DESC, Remark.MESSAGE_CONSTRAINTS); //invalid remark
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:92]:
```suggestion
        assertParseFailure(parser, "1" + INVALID_REMARK_DESC, Remark.MESSAGE_CONSTRAINTS); // invalid remark
```

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 7]
```diff
@@ -158,6 +163,12 @@ public void parse_oneFieldSpecified_success() {
         expectedCommand = new EditCommand(targetIndex, descriptor);
         assertParseSuccess(parser, userInput, expectedCommand);
 
+        //remark
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:166]:
```suggestion
        // remark
```

File: src/test/java/seedu/address/model/person/RemarkTest.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+class RemarkTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Remark(null));
+    }
+
+    @Test
+    public void constructor_invalidName_throwsIllegalArgumentException() {
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/model/person/RemarkTest.java:17]:
```suggestion
    public void constructor_invalidRemark_throwsIllegalArgumentException() {
```

File: src/test/java/seedu/address/model/person/RemarkTest.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        // invalid name
+        assertFalse(Remark.isValidRemark("^")); // only non-alphanumeric characters
+        assertFalse(Remark.isValidRemark("happy*")); // contains non-alphanumeric characters
+
+        // valid name
+        assertTrue(Remark.isValidRemark("happy")); // alphabets only
+        assertTrue(Remark.isValidRemark("12345")); // numbers only
+        assertTrue(Remark.isValidRemark("the 2nd son of Peter")); // alphanumeric characters
+        assertTrue(Remark.isValidRemark("Capital Land")); // with capital letters
+        assertTrue(Remark.isValidRemark("David Roger Jackson Ray Jr 2nd")); // long remarks
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/model/person/RemarkTest.java:36]: Should the `name`s be `remark`? Check the comments and method name.

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 10]
```diff
@@ -41,69 +44,91 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
...
+                VALID_REMARK, VALID_TAGS);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
     }
 
+    @Test
+    public void toModelType_invalidRemark_throwsIllegalValueException() {
+        JsonAdaptedPerson person =
+                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS,
+                        INVALID_REMARK, VALID_TAGS);
```
Comment (GnohChengYi) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:113]:
```suggestion
        JsonAdaptedPerson person = new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL,
                VALID_ADDRESS, INVALID_REMARK, VALID_TAGS);
```

Would it be better to make the style consistent?

**Review by GnohChengYi — CHANGES_REQUESTED** [review]
Hey, I like how clean your code is 👍 .
However, you forgot to change some keywords (especially in the comments) to `Remark`. 
Also, would it be better to have a space at the start of comments e.g. `// a comment` instead of `//a comment`, so that there are consistent with most other comments?

**Comment by ngchisern** [comment]
made changes in line with Cheng Yi's comments

**Review by GnohChengYi — APPROVED** [review]
LGTM!

**Pull request merged**